website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,218 @@
+
+## Why
+
+Why is compiling your dependencies desirable in the first place (vs. just your own code)?
+
+We can ship less code to users, since JavaScript has a [cost](https://medium.com/dev-channel/the-cost-of-javascript-84009f51e99e).
+
+## Assumptions
+
+- You can ship to modern browsers (don't have to support IE) or are able to send multiple types of bundles (one way being [checking for script`type=module`](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/)).
+- And/or the dependencies actually publish ES6+ instead of ES5/ES3.
+
+## Backing up a Bit: Talking About `preset-env`
+
+If you think about it, the argument for why this would be helpful is the same for why we eventually introduced `@babel/preset-env`.
+
+Babel used to be `6to5`, since it only converted from ES6 to ES5. Back then, the browser support for ES6 was almost non-existent so the idea of a JS compiler was an amazing contribution to our ecosystem.
+
+I went through with implementing the preset even though there was so much complexity since it seemed to solve multiple issues for us. I didn't realize how much it would move us forward now that it's finally [surpassed `preset-es2015`](https://npm-stat.com/charts.html?package=babel-preset-es2015&package=babel-preset-env&from=2016-11-01&to=2018-06-22) which is a nice accomplishment.
+
+When I think about Babel's role not just in moving the language itself forward but helping us develop websites better, I saw that we would eventually want to move past only compiling to ES5. Creating `preset-env` helps us align with the browsers since if we only compiled to ES5, no one would ever run native code in the browsers.
+
+The real difference is realizing that there will _always_ be a sliding window of support: whether it's the JavaScript language that continues to move forward with new syntax, the browsers catching up with the standard, or the companies/projects themselves.
+
+- You (your supported environments)
+- Browsers (Chrome, Firefox, Edge, Safari)
+- TC39/ECMAScript Proposals (and Babel implementations)
+
+Thus the need isn't just for `6to5` to be renamed to Babel because it compiles to `7to5`, but for Babel to change the implicit assumption it only targets ES5. With `@babel/preset-env`, you are able to write the latest JavaScript and target whichever browser (environment) you want!
+
+But assuming people even know it exists or have starting using it at all, it's mainly for your **application** code.
+
+## Your Code
+
+You have control over your own code to be able to take advantage of preset-env: write in ES6+ and target ES6+ browsers.
+
+However, this isn't necessarily the case for your dependencies (though you _should_ take ownership of them).
```

Comment:
The paranthetical here could be omitted or changed to convey that you can’t really do this well today, but you’ll want to, to get the benefits we talked about in your application code.